api_fhir_r4/converters/subscriptionConverter.py

```diff
@@ -0,0 +1,165 @@
+        fhir_subscription = {}
+        cls._build_fhir_id(fhir_subscription, imis_subscription)
+        cls._build_fhir_status(fhir_subscription, imis_subscription)
+        cls._build_fhir_end(fhir_subscription, imis_subscription)
+        cls._build_fhir_reason(fhir_subscription, imis_subscription)
+        cls._build_fhir_criteria(fhir_subscription, imis_subscription)
+        cls._build_fhir_error(fhir_subscription, imis_subscription)
+        cls._build_fhir_channel(fhir_subscription, imis_subscription)
+        return FHIRSubscription.parse_obj(fhir_subscription)
+
+    @classmethod
+    def to_imis_obj(cls, fhir_subscription, audit_user_id):
+        fhir_subscription = FHIRSubscription.parse_obj(fhir_subscription)
+        imis_subscription = {}
+        cls._build_imis_id(imis_subscription, fhir_subscription)
+        cls._build_imis_status(imis_subscription, fhir_subscription)
+        cls._build_imis_end(imis_subscription, fhir_subscription)
+        cls._build_imis_criteria(imis_subscription, fhir_subscription)
+        cls._validate_fhir_error(fhir_subscription)
+        cls._build_imis_channel_type(imis_subscription, fhir_subscription)
+        cls._build_imis_channel_endpoint(imis_subscription, fhir_subscription)
+        cls._build_imis_channel_header(imis_subscription, fhir_subscription)
+        return Subscription(**imis_subscription)
+
+    @classmethod
+    def get_fhir_code_identifier_type(cls):
+        return FHIRException("Subscription resource does not contain code identifier")
+
+    @classmethod
+    def _build_fhir_id(cls, fhir_subscription, imis_subscription):
+        fhir_subscription['id'] = str(imis_subscription.id)
+
+    @classmethod
+    def _build_fhir_status(cls, fhir_subscription, imis_subscription):
+        if imis_subscription.status in SubscriptionStatusMapping.to_fhir_status:
+            fhir_subscription['status'] = SubscriptionStatusMapping.to_fhir_status[imis_subscription.status]
+        else:
+            raise FHIRException(
+                cls._error_unknown_imis_value % {'attr': 'status', 'val': str(imis_subscription.status)})
+
+    @classmethod
+    def _build_fhir_end(cls, fhir_subscription, imis_subscription):
+        fhir_subscription['end'] = imis_subscription.expiring
+
+    @classmethod
+    def _build_fhir_reason(cls, fhir_subscription, imis_subscription):
+        fhir_subscription['reason'] = imis_subscription.criteria['resource_type']
+
+    @classmethod
+    def _build_fhir_criteria(cls, fhir_subscription, imis_subscription):
+        criteria = deepcopy(imis_subscription.criteria)
+        resource = criteria.pop('resource_type')
+        arguments = '&'.join(f'{key}={value}' for key, value in criteria.items())
+        fhir_subscription['criteria'] = f'{resource}/?{arguments}'
+
+    @classmethod
+    def _build_fhir_error(cls, fhir_subscription, imis_subscription):
+        error = None  # TODO Add that after the error logging is added to the model
```

Comment:
@malinowskikam You can move this TODO either above or below line with 'error = None'